.github/workflows/build_and_test.yml

```diff
@@ -23,7 +19,7 @@ env:
 jobs:
   build-neon:
     runs-on: dev
-    container: 369495373322.dkr.ecr.eu-central-1.amazonaws.com/rustlegacy:2746987948
+    container: 369495373322.dkr.ecr.eu-central-1.amazonaws.com/rust:latest
```

Comment:
Let's use a specific tag, not just the latest one.
Also, we can move the image tag to the workflow env and use it `${{ env.RUST_TAG }}` to not to repeat it for each job

Reply:
Would conflict with lifecycle rule on ECR: Only keep 32 non latest images - latest will be behind an approval step on the build repository.

Reply:
> Would conflict with lifecycle rule on ECR: Only keep 32 non latest images

Wow, this is weird from them. Should we consider using other images storages? [GitHub container registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry), for example?

I'd expect running the same code to produce the same output (or about the same), which is nearly impossible to do with using `latest` tags. It also makes it harder to debug CI failures (when images switch occasionally)

Reply:
No this is my added ECR rule, I guess I can put it to 128 non latest images but that would invalidate older pipelines, would that be preferred?

Reply:
Is it 128 across in total or per image?

Reply:
Per ECR repo, so per rust \ rustlegacy \ etc. Can go up much higher, but cost will increase.

Reply:
It depends on cost, but a stable build infrastructure is probably worth it

Reply:
What about this approach: https://github.com/neondatabase/build/pull/12 - then we have predictability but not inefficient levels of storage used

Reply:
Could you please explain what it does? 😄

Reply:
Check if any images have tag "pinned" - if so, keep at least 3. - continue with rule #3. As posted in that pull, not sure what makes sense for workflow to "tag" an image as such, as it is infrequent.

Reply:
Added the pinned tag, which for now is set manual. Will add this workflow to build readme

Reply:
Hmm, I think with `pinned` we'll have the same issues as with `latest` — workflow will pick up new images implicitly without any changes in the workflow itself.
I'm for having a new tag for each image version and letting the old one expire

Reply:
Yes, however comparing the balancing act of keeping enough images (inefficient) that are almost all identical so we can have one specific numbered image available, say `rust:2746987948` - versus a workflow that will highlight what needs to be done before someone promotes an image to become the newest `rust:pinned` I would say the later one offers more control.

Reply:
Ok, let's switch it back to the `latest` and discuss it separately (to not block this PR further)

Reply:
I got it on pinned now, which is outside of any automated workflow.